src/main/java/org/minions/devfund/bruno/html/HtmlValidator.java

```diff
@@ -0,0 +1,28 @@
+package org.minions.devfund.bruno.html;
+
+import java.util.Queue;
+import java.util.Stack;
+
+public class HtmlValidator {
+    public static Stack<HtmlTag> isValidHtml(Queue<HtmlTag> tags) {
+        Stack<HtmlTag> stack = new Stack<>();
+        for (HtmlTag tag : tags) {
+            if (tag.isOpenTag()) {
+                stack.push(tag);
+                continue;
+            }
+            if (tag.isSelfClosing()) {
+                continue;
+            }
+            if (stack.isEmpty()) {
+                return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1168)